packages/fxa-auth-server/lib/routes/subscriptions.js

```diff
@@ -0,0 +1,188 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict';
+
+const error = require('../error');
+const isA = require('joi');
+
+module.exports = (log, db, config, customs, oauthdb, subscriptionsBackend) => {
+  const {
+    subscriptions: {
+      productCapabilities = {},
+      clientCapabilities = {}
+    } = {}
+  } = config;
+
+  return [
+    {
+      // List subscription capabilities
+      method: 'GET',
+      path: '/subscriptions/capabilities',
+      options: {
+        auth: {
+          strategies: [
+            'sessionToken',
+            'oauthToken'
+          ]
+        },
+      },
+      handler: async function (request) {
+        log.begin('getSubscriptionCapabilities', request);
+        // TODO: is this where to get client_id?
```

Comment:
I want to use the ID of the client for which an OAuth token was issued to determine what subscription capabilities are visible.

How do I get that ID? Is it already in `request.auth.credentials`? Or will I have to bounce the token off an OAuth server API? I'm going to dig around to learn this myself, but someone might know this faster.

Reply:
If you have an `oauthToken` then yep, I think (?) it's a property on that. It won't be set if you have a `sessionToken` though.

Reply:
(btw, in case it wasn't clear, these strategies just end up returning instances of the token objects from `lib/tokens`, you can see how they're made with `makeCredentialFn` in `lib/server.js`)